### ISSUE TITLE
Add `ModifyAll` and `ModifyValidateAll`?

### ISSUE THREAD

**Issue by dariooddenino** [thomashoneyman/purescript-halogen-formless#31]
At the moment there's a `ValidateAll`, but the same is not possible with `Modify` and `ModifyValidate`.

Use case:
I have a form that gets populated with some initial values when the user clicks on a product.
Right now I have to manually call `modifyValidate_` for each form field in the parent component's `HandleInput` action.

It would be useful to be able to do something like
```
H.query unit $ F.modifyValidateAll_ { foo: 1, bar: "X", baz: false }
```

Not sure if it's better this way, or to have a `modifyValidateMultiple`, where one can just pass a subrecord instead of the whole one.

**Comment by thomashoneyman** [comment]
That's an interesting idea! Off the top of my head I can't think of a good reason not to allow modifying some subset of the form by providing all the necessary fields, or the same case for multiple validators all the way up to 'all validation'. That doesn't mean I'm 100% sure of an implementation, but it seems possible and useful.

I have a branch called `heterogeneous` which contains some serious changes to the library; I can try and wrap this in to those updates before releasing 0.3.0.

**Comment by thomashoneyman** [comment]
@dariooddenino I promise I haven't forgotten about this! I ran into some issues with heterogeneous regarding functional dependencies and haven't had the time to revise that branch and merge. I hope to do so sometime in the next week and then move on to this and adding the `set*` functions. With those out of the way, I'd like to make a 0.3.0 release. Thank you for your patience!

**Comment by thomashoneyman** [comment]
Closed by #35! I'm not able to support partial updates because doing so would require carrying around more type variables in the component; instead, I only offer modifying a single field or modifying all fields.
